src/protocols/bdx/AsyncTransferFacilitator.h

```diff
@@ -0,0 +1,108 @@
+ *
+ * This class does not define any methods for beginning a transfer or initializing the underlying TransferSession object (see
+ * Responder below. TODO: # 29334 - Add Initiator to AsyncFacilitator)
+ *
+ * For each BDX transfer, we will have an instance of AsyncTransferFacilitator facilitating the transfer.
+ */
+class AsyncTransferFacilitator : public Messaging::ExchangeDelegate
+{
+public:
+    AsyncTransferFacilitator() : mExchange(*this) {}
+    ~AsyncTransferFacilitator() override = default;
+
+    /**
+     * This method should be implemented to contain business-logic handling of BDX messages and other TransferSession events.
+     *
+     * @param[in] event An OutputEvent that contains output from the TransferSession object.
+     */
+    virtual void HandleAsyncTransferSessionOutput(TransferSession::OutputEvent & event) = 0;
+
+    /**
+     * This method returns the exchange context contained in the exchange holder member of this class.
+     */
+    chip::Messaging::ExchangeContext * GetExchangeContext();
+
+    // Inherited from ExchangeContext
+    CHIP_ERROR OnMessageReceived(chip::Messaging::ExchangeContext * ec, const chip::PayloadHeader & payloadHeader,
+                                 chip::System::PacketBufferHandle && payload) override;
+    void OnResponseTimeout(Messaging::ExchangeContext * ec) override;
+    void OnExchangeClosing(chip::Messaging::ExchangeContext * ec) override;
+
+protected:
+    // The transfer session coresponding to this AsynTransferFacilitator object.
+    TransferSession mTransfer;
+
+    // The Exchange holder that holds the exchange context used for the BDX messages.
+    Messaging::ExchangeHolder mExchange;
+};
+
+/**
+ * An AsyncTransferFacilitator that is initialized to respond to an incoming BDX transfer request.
+ *
+ * Provides a method for initializing the TransferSession member but still needs to be extended to implement
+ * HandleAsyncTransferSessionOutput. It is intended that this class will be used as a delegate for handling an unsolicited BDX
+ * message.
+ */
+class AsyncResponder : public AsyncTransferFacilitator
+{
+public:
+    /**
+     * Initialize the TransferSession state machine to be ready for an incoming transfer request.
+     *
+     * @param[in] exchangeCtx     The exchange context of the delegate
+     * @param[in] role            The role of the Responder: Sender or Receiver of BDX data
+     * @param[in] xferControlOpts Supported transfer modes (see TransferControlFlags)
+     * @param[in] maxBlockSize    The supported maximum size of BDX Block data
+     * @param[in] timeout         The chosen timeout delay for the BDX transfer
+     */
+    CHIP_ERROR PrepareForTransfer(Messaging::ExchangeContext * exchangeCtx, TransferRole role,
+                                  BitFlags<TransferControlFlags> xferControlOpts, uint16_t maxBlockSize,
+                                  System::Clock::Timeout timeout);
+
+    void ResetTransfer();
```

Comment:
this is called from MTROTAImageHandler ResetState where we reset state when an error occurs or transfer is complete.

Reply:
My point is: shouldn't that just correspond to destroying this object?  We used to have reset stuff because we reused objects, but now we have new-object-per-transfer.....

Reply:
agreed. yes.

Reply:
done.